### PR TITLE
ISPN-15124 Better describe SQL store limitations

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-15124

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/con_cache_store_sql.adoc [hunk 1]
```diff
@@ -22,15 +22,18 @@ Both SQL table and query stores:
 For composite keys and values, you must provide {brandname} with Protobuf schema (`.proto` files) that describe the keys and values.
 With {brandname} Server you can add schema through the {brandname} Console or Command Line Interface (CLI) with the [command]`schema` command.
 
-[NOTE]
+[WARNING]
 ====
-SQL cache stores do not support expiration when writes are allowed or segmentation.
+Since a SQL cache store is designed to be used with an existing database table it would not store any metadata, which includes expiration, segments and versioning metadata.
+Optimistic transactional cache and asynchronous cross-site replication won't work with the SQL store due to the lost of versions.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/con_cache_store_sql.adoc:28]: `s/lost/loss`

Comment (wburns) [inline documentation/src/main/asciidoc/topics/con_cache_store_sql.adoc:28]:
```suggestion
Optimistic transactional cache and asynchronous cross-site replication won't work with the SQL store due to not storing versions.
```
Just to make it a bit more explicit.

**Review by wburns — COMMENTED** [review]
LGTM

This PR is reminding me of the metadata storage characteristic I had suggested a while back so that we can validate configuration at runtime if a store will work with things like optimistic transactions or expiration.

**Comment by pruivo** [comment]
squashed and updated.

**Comment by domiborges** [comment]
Sorry for the delay. I added my commit with suggestions

**Comment by domiborges** [comment]
@pruivo can we merge this one?

**Comment by pruivo** [comment]
@domiborges yes please :+1: thanks!

**Pull request merged**